packages/protocol/contracts/tokenvault/ERC20Vault.sol

```diff
@@ -227,8 +243,18 @@ contract ERC20Vault is BaseVault {
 
         // If it's a bridged token
         if (bridgedToCanonical[token].addr != address(0)) {
+            // erc20NativeRegistry shall be address(0) on L1 and CAN be address(0) on L2 too, if
+            // there is no native support
```

Comment:
Would be cool if these changes would not be needed, but some things are not totally clear yet to me so I don't know.

Reply:
Please see this: https://github.com/taikoxyz/taiko-mono/pull/15250#discussion_r1405134515